vertx-sql-client/src/main/java/io/vertx/sqlclient/SqlClient.java

```diff
@@ -68,7 +68,7 @@
   <R> Future<SqlResult<R>> query(String sql, Collector<Row, ?, R> collector);
 
   /**
-   * Prepare and execute a query.
+   * Prepare a statement using the given SQL string and execute the prepared statement without any parameter, the asynchronous result is represented as {@link RowSet}.
```

Comment:
Statement will not always be prepared (because it might be reused), so instead:

`Execute the given {@code sql} string with a prepared statement ...`